kernel/syscall.c

```diff
@@ -5548,7 +5533,52 @@ static long _syscall(void* args_)
         case SYS_pselect6:
             break;
         case SYS_ppoll:
-            break;
+        {
+            struct pollfd* fds = (struct pollfd*)x1;
+            nfds_t nfds = (nfds_t)x2;
+            const struct timespec* timeout_ts = (const struct timespec*)x3;
+            const sigset_t* sigmask = (const sigset_t*)x4;
+            long timeout;
+            long ret;
+            struct timespec_buf buf;
+            sigset_t origmask;
+
+            _strace(
```

Comment:
I would wrap the call to ``_strace()`` in the following:

```
if (_trace_syscall(SYS_futimesat))
{
    _strace(...)
    
}
```

This will avoid the really inefficient _format_timespec() call when not tracing. On second, thought ignore for now since this pattern is used everywhere. I should create a PR just for that to clean up all of these.